src/overlays/actors/ovl_Bg_Bombwall/z_bg_bombwall.c

```diff
@@ -36,29 +61,192 @@ const ActorInit Bg_Bombwall_InitVars = {
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Bombwall/BgBombwall_Destroy.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Bombwall/func_8086EB5C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Bombwall/func_8086ED50.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Bombwall/func_8086ED70.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Bombwall/func_8086EDFC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Bombwall/func_8086EE40.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Bombwall/func_8086EE94.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Bombwall/BgBombwall_Update.s")
+static InitChainEntry sInitChain[] = {
+    ICHAIN_F32(uncullZoneForward, 1800, ICHAIN_CONTINUE), // unk f4
+    ICHAIN_F32(uncullZoneScale, 300, ICHAIN_CONTINUE),    // unk f8
+    ICHAIN_F32(uncullZoneDownward, 1000, ICHAIN_STOP),    // unk fc
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Bombwall/BgBombwall_Draw.s")
+Vec3s D_8086F010[] = { { 0x0028, 0x0055, 0x0015 }, { 0xFFD5, 0x006B, 0x000E }, { 0xFFFF, 0x008E, 0x000E },
+                       { 0xFFE5, 0x002C, 0x001B }, { 0x001C, 0x0018, 0x0014 }, { 0xFFD9, 0x0036, 0x0015 },
+                       { 0x0031, 0x0032, 0x0014 } };
+
+void BgBombwall_InitDynapoly(BgBombwall* this, GlobalContext* globalCtx) {
+    s32 pad;
+    s32 pad2;
+    ColHeader* colHeader;
+
+    colHeader = NULL;
+    DynaPolyInfo_SetActorMove(&this->dyna, 0);
+    DynaPolyInfo_Alloc(&D_050041B0, &colHeader);
+    this->dyna.dynaPolyId =
+        DynaPolyInfo_RegisterActor(globalCtx, &globalCtx->colCtx.dyna, &this->dyna.actor, colHeader);
+
+    if (this->dyna.dynaPolyId == 0x32) {
+        // Warning : move BG login failed
+        osSyncPrintf((const char*)"Warning : move BG 登録失敗(%s %d)(arg_data 0x%04x)\n", "../z_bg_bombwall.c", 0xF3,
+                     this->dyna.actor.params);
+    }
+}
+
+void BgBombwall_RotateVec(Vec3f* arg0, Vec3f* arg1, f32 arg2, f32 arg3) {
+    arg0->x = (arg1->z * arg2) + (arg1->x * arg3);
+    arg0->y = arg1->y;
+    arg0->z = (arg1->z * arg3) - (arg1->x * arg2);
+}
+
+void BgBombwall_Init(Actor* thisx, GlobalContext* globalCtx) {
+
+    s32 i;
+    s32 j;
+    Vec3f vecs[3];
+    Vec3f sp80;
+
+    s32 pad;
+    BgBombwall* this = THIS;
+
+    f32 sin = Math_Sins(this->dyna.actor.shape.rot.y);
+    f32 cos = Math_Coss(this->dyna.actor.shape.rot.y);
+
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    Actor_SetScale(thisx, 0.1f);
+
+    if (Flags_GetSwitch(globalCtx, this->dyna.actor.params & 0x3F) != 0) {
+        func_8086EE94(this, globalCtx);
+    } else {
+
+        BgBombwall_InitDynapoly(this, globalCtx);
+        this->unk_2A2 |= 2;
+        Collider_InitTris(globalCtx, &this->collider);
+        Collider_SetTris(globalCtx, &this->collider, this, &sTrisInit, &(this->colliderItems[0]));
```

Comment:
you want to pass the address of the array, not just the first item. while the mean the same thing, the types will line up
so just `this->colliderItems` is fine, you dont need & for functions or arrays